### DISCUSSION TITLE
Create magento_integration_tests DB by default

### DISCUSSION THREAD

**Discussion by Vinai** [wardenenv/warden#614] [Ideas]
In order to run integration tests, it would be helpful to pre-create an empty database `magento_integration_tests` and give the `magento` user access to it.

```
$ mysql -hdb -uroot -pmagento
Welcome to the MariaDB monitor.  Commands end with ; or \g.
Your MariaDB connection id is 397
Server version: 10.3.18-MariaDB-1:10.3.18+maria~bionic mariadb.org binary distribution

Copyright (c) 2000, 2018, Oracle, MariaDB Corporation Ab and others.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

MariaDB [(none)]> create database magento_integration_tests;
Query OK, 1 row affected (0.00 sec)
MariaDB [(none)]> grant all on magento_integration_tests.* to 'magento'@'%' identified by 'magento';
Query OK, 0 rows affected (0.00 sec)
```

I'm not sure how best to accomplish this with this setup, otherwise I'd be happy to create a PR.

**Comment by davidalger** [comment]
@Vinai Given the work @lbajsarowicz has done to implement and document (https://docs.warden.dev/configuration/magento2-testing.html) support for testing using an additional database container, what are you thoughts on this? Is there still a case for having a `magento_integration_tests` database created in the primary `db` container when it starts up?

**Comment by Vinai** [comment]
It took me a while to get back to this, but unfortunately currently I'm unable to use the additional container, see issue #136 

The only downside of using the temp db is that when returning to a project after some time, the initial db will need to be re-initialized on the first run even if `TESTS_CLEANUP` is enabled.
It's basically a tradeoff between a quick re-entry into a project, or faster test runs.
Personally, I would like to have the option to switch either on.

**Comment by davidalger** [comment]
That makes sense, and appreciate the input. I'll keep this open until it can be implemented. Another probable scenario is some may not have enough available memory to run the integration db in-memory (easy for me to forget since both my Mac and Dell workstations have 64 GB in them; laptops frequently have less available to allocate though)

[a comment by an automated account is not shown]

**Comment by Vinai** [comment]
This issue still is relevant. I'm always using the regular database for integration testing instead of the tmpfs mounted one because the speed difference is negligible and I don't want to wait for the initial install after starting the container.
